homeassistant/components/solax/sensor.py

```diff
@@ -102,32 +113,44 @@ async def async_setup_entry(
     resp = await api.get_data()
     serial = resp.serial_number
     version = resp.version
+    last_reset = dt_util.start_of_local_day()
     endpoint = RealTimeDataEndpoint(hass, api)
+    for sensor, (idx, measurement) in api.inverter.sensor_map().items():
+        description = SENSOR_DESCRIPTIONS[
+            (measurement.unit, measurement.is_monotonic, measurement.storage)
+        ]
+
+        uid = f"{serial}-{idx}"
+        device = Inverter(
+            api.inverter.manufacturer,
+            uid,
+            serial,
+            version,
+            sensor,
+            description.native_unit_of_measurement,
+            description.state_class,
+            description.device_class,
+            last_reset if measurement.resets_daily else None,
+        )
+        if measurement.resets_daily:
+            entry.async_on_unload(
+                async_track_time_change(
+                    hass=hass,
+                    action=device.async_listen_for_midnight,
+                    hour=0,
+                    minute=0,
+                    second=0,
```

Comment:
How do we know that the device and HA use the same time zone?

Reply:
We don't...

The dongle in the inverter does allow the user to configure a time zone. (To anyone reading along with a hybrid inverter and a battery, I know the app handles daylight savings time correctly, but does the local API this extension uses also handles it correctly? In other words: do the daily totals reset to 0 that corespond to local midnight in both winter and summer? I only have a X1 mini which goes into a standby mode from sunset to sunrise.)

Should I make the time zone configurable?
And if so, should I create a notification with id `RECONFIGURE_NOTIFICATION_ID` and fail to initialise? Or a notification and default to the local timezone? Or ... something else entirely?

Also, now that there is a coordinator, calling `async_track_time_change` for each daily resetting entity seems like it could be a code smell. Is calling `coordinator.async_set_updated_data()` and overriding `InverterSensorEntity._handle_coordinator_update()` so it handles the entity's key not being present in the updated data better? Or just different?

Reply:
> Is calling `coordinator.async_set_updated_data()` and overriding `InverterSensorEntity._handle_coordinator_update()` so it handles the entity's key not being present in the updated data better? Or just different?

That seems better

> The dongle in the inverter does allow the user to configure a time zone. (To anyone reading along with a hybrid inverter and a battery, I know the app handles daylight savings time correctly, but does the local API this extension uses also handles it correctly? In other words: do the daily totals reset to 0 at a time that corespond to local midnight in both winter and summer? I only have a X1 mini which goes into a standby mode from sunset to sunrise.)

Is there any way to get the time zone data out of the device?

Reply:
I think we stuck with the assumption that the time zones match, and its something we have to live with in other integrations as well. Thats usually the case but it better if we don't have to make that assumption.

Reply:
I can set the timezone on the device via Solax Portal site and it matches my personal timezone, just like with the HA device. But good to check whether we might run into issues with time zone mismatches.